File: app/src/main/java/com/example/pokedex/data/remote/network/ApiService.java
package com.example.pokedex.data.remote.network;

import com.example.pokedex.data.remote.response.DetailPokemonResponse;
import com.example.pokedex.data.remote.response.PokemonResponse;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface ApiService {
    @GET("api/v2/pokemon")
    Call<PokemonResponse> getPokemonList(
            @Query("offset") int offset,
            @Query("limit") int limit);

    @GET("api/v2/pokemon/{id}")
    Call<DetailPokemonResponse> getDetailPokemon(@Path("id") int id);


}

File: core/src/main/java/com/example/data/remote/network/ApiService.kt
package com.example.data.remote.network

import retrofit2.http.GET
import com.example.data.remote.response.PokemonResponse
import com.example.data.remote.response.DetailPokemonResponse
import io.reactivex.Observable
import io.reactivex.Single
import retrofit2.Call
import retrofit2.http.Path
import retrofit2.http.Query

interface ApiService {
  @GET("api/v2/pokemon") fun getPokemonList(
    @Query("offset") offset: Int,
    @Query("limit") limit: Int
  ): Call<PokemonResponse>

  @GET("api/v2/pokemon/{id}")
  fun getDetailPokemon(@Path("id") id: Int): Call<DetailPokemonResponse>

  @GET("api/v2/pokemon") fun getPokemonListPaging(
    @Query("offset") offset: Int,
    @Query("limit") limit: Int
  ): Single<PokemonResponse>

  @GET("api/v2/pokemon") fun getPokemonListObservable(
    @Query("offset") offset: Int,
    @Query("limit") limit: Int
  ): Observable<PokemonResponse>
}